create a manual for archive data on the NIRD research archive. (#81)
NorwegianVeterinaryInstitute/nvi-documentation: docs/source/tutorials/13-nra_tutorial.md

# How to archive a dataset in the NIRD Research Data Archive

**Author:** Thomas Haverkamp   
**Purpose:** This tutorial walks through the process of submitting an old dataset to the NIRD Research Data Archive (NRA), based on a real test submission (the "Koksfri Kylling" dataset). Use it as a step-by-step guide the next time you need to archive a dataset. Or just read through it, to get familiar with the process. The real user manual for the NRA is following the same flow, but omits a lot of details so it get confusing.

## What is the NIRD archive?

> The NIRD Research Data Archive is a place to store datasets — for example old project data — that are no longer actively needed at the institute but should be preserved and made findable.

Reviewer: Maybe highlight that it is not for sequencing data, but for analysis scripts and results?

Author: Changed it

- Archive: <https://archive.sigma2.no/>
- User guide: <https://documentation.sigma2.no/nird_archive/user-guide.html>
- Archive workflow overview: <https://documentation.sigma2.no/nird_archive/archive-workflow.html>

## Before you start

- **Get access to the archive.** Log in at <https://data.archive.sigma2.no/user/login>. Access is possible via Feide, but Feide login may need to be activated for your institute — ask your local IT/NRIS contact if it isn't working yet.

  <img src="img/01-feide-login.png" alt="image" width="50%" height="auto">

- **Identify the dataset** you want to archive, and note where it is stored (e.g. on NIRD `datalake`/`datapeak`) and its size. 
- **Ask permission from the data owner(s)/original project members** before archiving their data, especially if you are not the sole owner. Also check with them whether the data should also be submitted to a public repository such as NCBI/ENA (for sequence data), since this may be a separate step.

> - **Clean up the dataset**. Remove any intermediate analysis files that you could recreate by using your analysis / pipelines scripts. These files should not be archived. Archive your final datasets. And don't forget to create a file with the md5sum or sha256sum for the datasets to the folder where your data is. In that we corruption of files can be tracked. 

Reviewer: ```suggestion
- **Clean up the dataset**. Remove any intermediate analysis files that you could recreate by using your analysis / pipelines scripts. These files should not be archived. Archive your final datasets. And don't forget to create a file with the md5sum or sha256sum for the datasets to the folder where your data is. File corruption can be tracked by using these values. 
```

> - **Clean up the dataset**. Remove any intermediate analysis files that you could recreate by using your analysis / pipelines scripts. These files should not be archived. Archive your final datasets. And don't forget to create a file with the md5sum or sha256sum for the datasets to the folder where your data is. In that we corruption of files can be tracked. 

Reviewer: Perhaps write how to archive? Later we can connect this to ATLAS docs

Author: done

## Step 2: Fill in the metadata

Next, fill in the metadata for the dataset:

- **Title** – a short, descriptive title.
- **Description** – explain what the dataset contains, the background/context of the project, and how the data was generated/analyzed. Include a link to the project or a relevant publication if available.
- **Keywords** – add a few keywords that describe the dataset (e.g. subject organism, method, sample type).
- **Geospatial metadata fields (CF standard names, GEMET, ISO)** – these three categories are for describing *geospatial* datasets specifically. **They are not relevant for most of our datasets** and can be left empty. If you do want to add something under ISO, a general topical keyword (e.g. "farming") is fine.

  ![The geospatial metadata fields (CF standard names, GEMET, ISO) — not relevant for non-geospatial datasets](img/03-geospatial-metadata-fields.png)

- **Subject** – choose the relevant subject area (e.g. "Natural sciences").
- **Contact point** – add your institute (e.g. NVI) with your own email address as the contact.
- **Creators** – add yourself and any co-creators of the dataset, with their email addresses.

> - **Contributors** – leave blank unless relevant (e.g. an external partner organization).

Reviewer: Are contributors here co-authors of the code, or only institutions?

Author: I actually have no idea. The documentation for the NIRD archive is in that sense to nt very clear

- **Data owner** – set to your institute.
- **License** – choose an appropriate license (e.g. Creative Commons Attribution 4.0).
- **Access rights** – e.g. "Public".
- **Temporal coverage** – the start and end date of the project the data comes from.
- **Related URL** – a link to the project page, if available.

> - **Version note** – add a short note describing which version of the dataset this is.

Reviewer: Is this connected to GIT somehow?

Author: No versions of the data and git are not connect. You can not use git to update the data.. You update the data by indicating by selecting a new version in the portal.

- A `README` file describing the contents
- A checksum file (e.g. `.sha256`) so the integrity of the upload can be verified

Note the full path to the folder where these files are stored — you will need it for the manifest file in the next step.

## Step 5: Create the manifest file

Uploading data via the **NIRD Project** option works through a **manifest file**: a text file that lists exactly which files should be uploaded to the archive.

1. **Find the dataset identifier.** Go to your profile on the archive website and open the dataset you just created (draft). The dataset's unique identifier (a UUID) is shown there — this is what you need for the manifest file name.

   ![Finding your dataset in your profile](img/05-dataset-in-profile.png)

   ![The dataset identifier (UUID) shown on the dataset page](img/06-dataset-identifier.png)

2. **Create the manifest file** in your home folder on NIRD (e.g. `/nird/home/<username>`). The file must be named:

   ```
   .import-archive_<dataset-identifier>
   ```

   For example, if the identifier is `12ade14d-9b39-4995-a44f-e63ab59eb730`, the file would be named:

   ```
   .import-archive_12ade14d-9b39-4995-a44f-e63ab59eb730
   ```

> 3. **List the files to upload** inside the manifest file, one full path per line. Add an extra `/` in the path right before the part you actually want copied — this ensures only the dataset itself (not the entire directory structure above it) is copied to NIRD. For example:

Reviewer: Here, adding only one more slash is mentioned, but there are two extra in the example below. Which is correct?

Author: According to the manual it should actually three `/`. I see that it is confusing here. I will make it two slashes in the text

Author: done
